### PR TITLE
Spring quickstart

### PR DESCRIPTION
**Pull request by slaskawi**
Hi!

Please take a look at the implementation of Spring integration Quickstart. I'll backport it to the production branch as soon as upstream gets merged.

Implementation details:
* Used Spring boot to simplify Spring setup.
* Created Ember JS frontend which communicates with backend via REST interfaces (based on Spring MVC).
* Used Patternfly look and feel to be consistent with Red Hat brand.

Thanks!
Sebastian

### CODE REVIEW

**Comment by slaskawi** [comment]
Hey @mgencur @tristantarrant !

CI passes with this PR, could you please take a look at the content? If everything is fine - I would like to push it and then prepare a backport for 6.4.

Thanks!

File: spring/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+Target Product: JDG
+Product Versions: JDG 6.x
+Source: <https://github.com/infinispan/jdg-quickstart>
+
+What is it?
+-----------
+
+Spring-jdg is an example that shows how to use Infinispan or JBoss Data Grid together with Spring.
+
+In order to simplify the Spring Bootstrapping this project uses Spring Boot.
```
Comment (tristantarrant) [inline spring/README.md]: s/Bootstrapping/bootstrapping/

File: spring/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+_NOTE: The following build command assumes you have configured your Maven user settings. If you have not, you must
+include Maven setting arguments on the command line. See [Build and Deploy the Quickstarts](../../README.md#build-and-deploy-the-quickstarts)
+for complete instructions and additional options._
+
+1. Type this command to build it and start local Tomcat Embedded server:
+
+        mvn clean package spring-boot:run
+
+2. Navigate to <http://localhost:8080> and follow instructions in about page.
```
Comment (tristantarrant) [inline spring/README.md]: follow the instructions

File: spring/src/main/java/org/jboss/as/quickstarts/datagrid/spring/bootstrap/ServletInitializer.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.boot.context.web.SpringBootServletInitializer;
+
+/**
+ * Spring MVC configuration.
+ *
+ * @author Sebastian Laskawiec
+ */
+public class ServletInitializer extends SpringBootServletInitializer {
+
```
Comment (tristantarrant) [inline spring/src/main/java/org/jboss/as/quickstarts/datagrid/spring/bootstrap/ServletInitializer.java]: Weird indentation

File: spring/src/main/java/org/jboss/as/quickstarts/datagrid/spring/core/client/ClientCache.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * Handles direct cache access.
+ *
+ * @author Sebastian Laskawiec
+ */
+public class ClientCache {
+
+    private Cache cache;
+    private AdvancedCache<Object, Collection<Client>> nativeCache;
```
Comment (tristantarrant) [inline spring/src/main/java/org/jboss/as/quickstarts/datagrid/spring/core/client/ClientCache.java]: Why is the key an Object, can't we qualify it better ?

File: spring/README.md (hunk 1 above)
Comment (slaskawi) [inline spring/README.md]: done

File: spring/README.md (hunk 2 above)
Comment (slaskawi) [inline spring/README.md]: done

File: spring/src/main/java/org/jboss/as/quickstarts/datagrid/spring/bootstrap/ServletInitializer.java (hunk 3 above)
Comment (slaskawi) [inline spring/src/main/java/org/jboss/as/quickstarts/datagrid/spring/bootstrap/ServletInitializer.java]: converted tabs to spaces

File: spring/src/main/java/org/jboss/as/quickstarts/datagrid/spring/core/client/ClientCache.java (hunk 4 above)
Comment (slaskawi) [inline spring/src/main/java/org/jboss/as/quickstarts/datagrid/spring/core/client/ClientCache.java]: Switched to `AdvancedCache<String, Collection<Client>>`

**Comment by slaskawi** [comment]
Comments addressed. 

I also changed the following things:
* removed JFairy dependency (it would bite us later on)
* removed Actuator and WebSockets from the classpath (Wildfly deployment speedup + removed warnings)
* updated readme information and suggested running it on WildFly or EAP
* Updated Quickstart version in `pom.xml`

**Comment by slaskawi** [comment]
@mgencur, @tristantarrant Could you please take a look at this?

**Comment by slaskawi** [comment]
Integrating

**Comment by slaskawi** [comment]
Integrated

**Pull request closed without merging**